#!/usr/bin/env python
PACKAGE = "engagement_system"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("proxemics_weight",    double_t,    0, "Weight for proxemics computation", .5,  0, 1)
gen.add("proxemics_epsilon",    double_t,    0, "Epsilon for proxemics shifting", 1.5,  0, 5)
gen.add("gaze_weight",    double_t,    0, "Weight for gaze computation", .5,  0, 1)

# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)
# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                        gen.const("Medium",     int_t, 1, "A medium constant"),
#                        gen.const("Large",      int_t, 2, "A large constant"),
#                        gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                      "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "engagement_system", "Eng"))